### PR TITLE
SyncTube presence

### PR DESCRIPTION
**Pull request by N0chteil**
Proof:
Screenshot_293
Screenshot_294

resolves: #3158

### CODE REVIEW

File: websites/S/SyncTube/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    {
+      "id": "logo",
+      "title": "Icon",
+      "icon": "fad fa-icons",
+      "value": 0,
+      "values": ["With text", "Icon"]
+    }
+  ]
+}
```
Comment (Bas950) [inline websites/S/SyncTube/dist/metadata.json:47]: please put privacy mode at the top

File: websites/S/SyncTube/presence.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  if(urlpath[2]) {
+    presenceData.details = (set_privacy) ? "In room" : document.querySelector("div.roomName.noselect").textContent;
+    presenceData.state = (set_privacy) ? "" : document.querySelector("div.userCount.noselect").textContent;
+
+    if(set_showButtons) {
+      presenceData.buttons = [
+        {
+          label: 'Join room',
+          url: window.location.href // https://sync-tube.de/rooms/THE_ROOM
+        }
```
Comment (Bas950) [inline websites/S/SyncTube/presence.ts:31]: remove comment (useless comment)

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/S/SyncTube/presence.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      presenceData = {
+            largeImageKey: (set_logo === 0) ? "logo" : "logo2"
+          },
+      urlpath = window.location.pathname.split("/");
+
+if (set_timeElapsed) {
+  presenceData.startTimestamp = browsingStamp;
+}
+if(!urlpath[1]) {
+  presenceData.details = "Home";
```
Comment (ririxidev) [inline websites/S/SyncTube/presence.ts:20]: You don't need brackets for one line code.

File: websites/S/SyncTube/presence.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          url: window.location.href // https://sync-tube.de/rooms/THE_ROOM
+        }
+      ];
+    }
+  } else {
+    presenceData.details = "Browsing rooms";
+  }
+} else {
+  presenceData.details = "Other";
+}
```
Comment (ririxidev) [inline websites/S/SyncTube/presence.ts:39]: You don't need brackets for one line code.

File: websites/S/SyncTube/presence.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+if (set_timeElapsed) {
+  presenceData.startTimestamp = browsingStamp;
+}
+if(!urlpath[1]) {
+  presenceData.details = "Home";
+} else if(urlpath[1] === 'rooms') {
+  if(urlpath[2]) {
+    presenceData.details = (set_privacy) ? "In room" : document.querySelector("div.roomName.noselect").textContent;
+    presenceData.state = (set_privacy) ? "" : document.querySelector("div.userCount.noselect").textContent;
```
Comment (ririxidev) [inline websites/S/SyncTube/presence.ts:24]: Never set state/details whatever to "" nor null nor undefined.
```suggestion
    presenceData.details = (set_privacy) ? "In room" : document.querySelector("div.roomName.noselect").textContent;
    if (!set_privacy) presenceData.state = document.querySelector("div.userCount.noselect").textContent;
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @N0chteil!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

File: websites/S/SyncTube/presence.ts (hunk 2 above)
Comment (N0chteil) [reply websites/S/SyncTube/presence.ts:31]: https://github.com/PreMiD/Presences/pull/3412#discussion_r606606723

**Review by ririxidev — APPROVED** [review]
lgtm

**Review by StrikerFRFX — APPROVED** [review]
lgtm

**Pull request merged**